app/models/merchant.rb

```diff
@@ -37,4 +37,18 @@ def top_5_items
         .limit(5)
         .sum('quantity * invoice_items.unit_price')
   end
+
+  def self.top_5_by_revenue
+    joins(:transactions)
+    .where(transactions: {result: 1})
```

Comment:
I think we need to add a condition for which invoices to consider for the Revenue (we wouldn't want in progress or cancelled invoices.)

Reply:
agreed good eye Mostafa, our query king!

Reply:
Do you think we'd need that even though we're already filtering for successful transactions? I'm not super clear on what the relationship between transaction status and invoice status would be